### PR TITLE
Update grpc 1.47.0 patch + add additional dependencies for grpc upgrade

### PR DESCRIPTION
**Pull request by aranguyen**

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -659,6 +659,15 @@ dist_http_archive(
 # grpc_extra_deps().
 load("@com_github_grpc_grpc//bazel:grpc_deps.bzl", "grpc_deps")
 
+# for patching the "com_github_cncf_udpa" deps loaded by grpc_deps
+dist_http_archive(
```
Comment (meteorcloudy) [inline WORKSPACE:663]: You should move this before `load("@com_github_grpc_grpc//bazel:grpc_deps.bzl", "grpc_deps")` to make sure our definitions can take precedence.

**Review by meteorcloudy — CHANGES_REQUESTED** [review]

**Comment by meteorcloudy** [comment]
To fix the src test, you'll have to define those `src` targets for com_github_cncf_udpa and com_envoyproxy_protoc_gen_validate:
https://github.com/bazelbuild/bazel/blob/4e54f6bea0d5daca0f0953a408559f510237bdc5/third_party/protobuf/BUILD#L9

**Review by aranguyen — COMMENTED** [review]
looks good. Thanks!

**Review by meteorcloudy — CHANGES_REQUESTED** [review]

**Comment by aranguyen** [comment]
yikes...we're gonna need to pull in upd as well https://github.com/grpc/grpc/blob/2717aa8c9b768de99c811628af4e8c30d6c2da4f/bazel/grpc_deps.bzl#L349

**Comment by aranguyen** [comment]
closing this to create a new one that is hopefully less messy and does not contain others' commits

**Pull request closed without merging**